pkg/sysfs/syspath.go

```diff
@@ -0,0 +1,320 @@
+		dependents.Partitions = partitions
+	}
+
+	// get the holder devices
+	if holders, ok := s.getHolders(); ok {
+		dependents.Holders = append(dependents.Holders, holders...)
+	}
+
+	// get the slaves
+	if slaves, ok := s.getSlaves(); ok {
+		dependents.Slaves = append(dependents.Slaves, slaves...)
+	}
+
+	// adding /dev prefix
+	if len(dependents.Parent) != 0 {
+		dependents.Parent = "/dev/" + dependents.Parent
+	}
+
+	// adding /devprefix to partition, slaves and holders
+	dependents.Partitions = addDevPrefix(dependents.Partitions)
+	dependents.Holders = addDevPrefix(dependents.Holders)
+	dependents.Slaves = addDevPrefix(dependents.Slaves)
+
+	return dependents, nil
+}
+
+// GetLogicalBlockSize gets the logical block size, the caller should handle if 0 LB size is returned
+func (s Device) GetLogicalBlockSize() (int64, error) {
+	logicalBlockSize, err := readSysFSFileAsInt64(s.sysPath + "/queue/logical_block_size")
+	if err != nil {
+		return 0, err
+	}
+	return logicalBlockSize, nil
+}
+
+func (s Device) GetPhysicalBlockSize() (int64, error) {
+	physicalBlockSize, err := readSysFSFileAsInt64(s.sysPath + "/queue/physical_block_size")
+	if err != nil {
+		return 0, err
+	}
+	return physicalBlockSize, nil
+}
+
+func (s Device) GetHardwareSectorSize() (int64, error) {
+	hardwareSectorSize, err := readSysFSFileAsInt64(s.sysPath + "/queue/hw_sector_size")
+	if err != nil {
+		return 0, err
+	}
+	return hardwareSectorSize, nil
+}
+
+func (s Device) GetDriveType() (string, error) {
+	rotational, err := readSysFSFileAsInt64(s.sysPath + "/queue/rotational")
+	if err != nil {
+		return "", err
+	}
+
+	if rotational == 1 {
+		return blockdevice.DriveTypeHDD, nil
+	} else if rotational == 0 {
+		return blockdevice.DriveTypeSSD, nil
+	}
+	return "", fmt.Errorf("undefined rotational value %d", rotational)
+}
+
+func (s Device) GetCapacityInBytes() (int64, error) {
+	// The size (/size) entry returns the `nr_sects` field of the block device structure.
+	// Ref: https://elixir.bootlin.com/linux/v4.4/source/fs/block_dev.c#L1267
+	//
+	// Traditionally, in Unix disk size contexts, “sector” or “block” means 512 bytes,
+	// regardless of what the manufacturer of the underlying hardware might call a “sector” or “block”
+	// Ref: https://elixir.bootlin.com/linux/v4.4/source/fs/block_dev.c#L487
+	//
+	// Therefore, to get the capacity of the device it needs to always multiplied with 512
+	numberOfBlocks, err := readSysFSFileAsInt64(s.sysPath + "/size")
+	if err != nil {
+		return 0, err
+	} else if numberOfBlocks == 0 {
+		return 0, fmt.Errorf("block count reported as zero")
+	}
+	return numberOfBlocks * sectorSize, nil
+
+}
+
+// GetDeviceType gets the device type, as shown in lsblk
+// devicetype should be prefilled by udev probe (DEVTYPE) as disk/part for this to work
+//
+// Ported from https://github.com/karelzak/util-linux/blob/master/misc-utils/lsblk.c
+func (s Device) GetDeviceType(devType string) (string, error) {
```

Comment:
@richardelling  Can you review this piece of code. Its actually a port of `lsblk` from util-linux.